src/controllers/hid/hidcontroller.h

```diff
@@ -60,45 +57,79 @@ class HidControllerJSProxy : public ControllerJSProxy {
               m_pHidController(m_pController) {
     }
 
-    Q_INVOKABLE void send(const QList<int>& data, unsigned int length = 0) override {
-        m_pHidController->send(data, length);
+    /// @brief Sends HID OutputReport with hard coded ReportID 0 to HID device
+    ///        This function only works with HID devices, which don't use ReportIDs
+    /// @param dataList Data to send as list of bytes
+    /// @param length Unused optional argument
+    Q_INVOKABLE void send(const QList<int>& dataList, unsigned int length = 0) override {
+        // This function is only for class compatibility with the (midi)controller
+        Q_UNUSED(length);
+        this->send(dataList, 0, 0);
+    }
+
+    /// @brief Sends HID OutputReport to HID device
+    /// @param dataList Data to send as list of bytes
+    /// @param length Unused but mandatory argument
+    /// @param reportID 1...255 for HID devices that uses ReportIDs - or 0 for devices, which don't use ReportIDs
+    /// @param resendUnchangedReport If set, the report will also be send, if the data are unchanged since last sending
+    Q_INVOKABLE void send(const QList<int>& dataList,
```

Comment:
You have changed the signature now from 

```
Q_INVOKABLE void send(const QList<int>& data, unsigned int length, unsigned int reportID) {
```

```
Q_INVOKABLE void send(const QList<int>& dataList, unsigned int length, quint8 reportID, bool esendUnchangedReport = false)
```

But the function is deprecated anyway. Can this cause any compatibility issues? If in doubt I would prefer no to touch it.

Reply:
I tested it with the unmodified common-hid-packet-parser.js and it works flawless, with and without the optional argument.

Reply:
Thank you for the double check!